server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/IdentitySetUpdatePerformer.scala

```diff
@@ -0,0 +1,146 @@
+      .left.map({
+      case e: IllegalArgumentException => e
+      case e => new IllegalArgumentException(e)
+    }).map(uuid => IdentityId(uuid))
+  }
+
+  object IdentityUpdateParseException {
+    def from(errors: collection.Seq[(JsPath, collection.Seq[JsonValidationError])]): IdentityUpdateParseException = {
+      val setError: SetError = errors.head match {
+        case (path, Seq()) => SetError.invalidArguments(SetErrorDescription(s"'$path' property in Identity object is not valid"))
+        case (path, Seq(JsonValidationError(Seq("error.path.missing")))) =>
+          SetError.invalidArguments(SetErrorDescription(s"Missing '$path' property in Identity object"))
+        case (path, Seq(JsonValidationError(Seq(message)))) => SetError.invalidArguments(SetErrorDescription(s"'$path' property in Identity object is not valid: $message"))
+        case (path, _) => SetError.invalidArguments(SetErrorDescription(s"Unknown error on property '$path'"))
+      }
+      IdentityUpdateParseException(setError)
+    }
+  }
+
+  case class IdentityUpdateParseException(setError: SetError) extends IllegalArgumentException
+
+  object IdentityUpdate {
+    private val serverSetProperty: Set[String] = Set("id", "mayDelete", "email")
+    private val assignableProperties: Set[String] = Set("name", "replyTo", "bcc", "textSignature", "htmlSignature")
+    private val knownProperties: Set[String] = assignableProperties ++ serverSetProperty
+
+    def validateProperties(jsObject: JsObject): Either[IdentityUpdateParseException, JsObject] =
+      (jsObject.keys.intersect(serverSetProperty), jsObject.keys.diff(knownProperties)) match {
+        case (_, unknownProperties) if unknownProperties.nonEmpty =>
+          Left(IdentityUpdateParseException(SetError.invalidArguments(
+            SetErrorDescription("Some unknown properties were specified"),
+            Some(Properties.toProperties(unknownProperties.toSet)))))
+        case (specifiedServerSetProperties, _) if specifiedServerSetProperties.nonEmpty =>
+          Left(IdentityUpdateParseException(SetError.invalidArguments(
+            SetErrorDescription("Some server-set properties were specified"),
+            Some(Properties.toProperties(specifiedServerSetProperties.toSet)))))
+        case _ => scala.Right(jsObject)
+      }
+  }
+}
+
+class IdentitySetUpdatePerformer @Inject()(identityRepository: IdentityRepository) {
+  def update(request: IdentitySetRequest, mailboxSession: MailboxSession): SMono[IdentitySetUpdateResults] =
+    SFlux.fromIterable(request.update.getOrElse(Map()))
+      .concatMap {
+        case (unparsedId, json) =>
+          val either: Either[Exception, SMono[UpdateResult]] = for {
+            identityId <- unparsedId.parse
+            updateRequest <- parseRequest(json)
+          } yield {
+            update(identityId, updateRequest, mailboxSession)
+              .onErrorResume((error: Throwable) => SMono.just[UpdateResult](UpdateFailure(unparsedId, error)))
```

Comment:
```suggestion
              .onErrorResume(error => SMono.just[UpdateResult](UpdateFailure(unparsedId, error)))
```

?

Reply:
the new version of IDE suggests it, maybe it is more modern?

Reply:
Yes I prefer implicit typing in lambdas.